credentials/generate-revocation-set.py

```diff
@@ -44,20 +45,144 @@ class RevocationType(Enum):
 TEST_NODE_URL_REST = "https://on.test-net.dcl.csa-iot.org"
 
 
-def use_dcld(dcld, production, cmdlist):
-    return [dcld] + cmdlist + (['--node', PRODUCTION_NODE_URL] if production else [])
-
-
 def extract_single_integer_attribute(subject, oid):
     attribute_list = subject.get_attributes_for_oid(oid)
 
     if len(attribute_list) == 1:
-        if attribute_list[0].value.isdigit():
-            return int(attribute_list[0].value)
+        return int(attribute_list[0].value, 16)
 
     return None
 
 
+class DCLDClient:
+    '''
+    A client for interacting with DCLD using either the REST API or command line interface (CLI).
+
+    '''
+
+    def __init__(self, use_rest: bool, dcld_exe: str, production: bool, rest_node_url: str):
+        '''
+        Initialize the client
+
+        use_rest: bool
+            Use RESTful API with HTTPS against `rest_node_url`
+        dcld_exe: str
+            Path to `dcld` executable
+        production: bool
+            Use MainNet DCL URL with dcld executable
+        rest_node_url: str
+            RESTful API URL
+        '''
+
+        self.use_rest = use_rest
+        self.dcld_exe = dcld_exe
+        self.production = production
+        self.rest_node_url = rest_node_url
+
+    def build_dcld_command_line(self, cmdlist: list[str]) -> list[str]:
+        '''
+        Build command line for `dcld` executable.
+
+        Parameters
+        ----------
+        cmdlist: list[str]
+            List of command line arguments to append to some predefined arguments
+
+        Returns
+        -------
+        list[str]
+            The complete command list including the DCLD executable and node option if in production
+        '''
+
+        return [self.dcld_exe] + cmdlist + (['--node', PRODUCTION_NODE_URL] if self.production else [])
+
+    def get_dcld_cmd_output_json(self, cmdlist: list[str]) -> dict:
+        '''
+        Executes a DCLD CLI command and returns the JSON output.
+
+        Parameters
+        ----------
+        cmdlist: list[str]
+            List of command line arguments to append to some predefined arguments
+
+        Returns
+        -------
+        dict
+            The JSON output from the command
+        '''
+
+        # Set the output as JSON
+        subprocess.Popen([self.dcld_exe, 'config', 'output', 'json'])
+
+        cmdpipe = subprocess.Popen(self.build_dcld_command_line(cmdlist),
```

Comment:
should we have `check=True` here?

Reply:
[Popen()](https://docs.python.org/3/library/subprocess.html#subprocess.Popen) do not have the `check=` argument